lib/version_checker.rb

```diff
@@ -10,58 +10,50 @@ class HttpError < RuntimeError
 
 class VersionChecker
   def print_version_discrepancies
-    discrepancies_found = false
-
-    fetch_all_govuk_gemspecs.each do |gemspec|
-      rubygems_version = fetch_rubygems_version(gemspec.name)
-      unless gemspec.version == rubygems_version
-        discrepancies_found = true
-        puts "Version mismatch for '#{gemspec.name}':"
-        puts "  Version on RubyGems: #{rubygems_version}"
-        puts "  Version on GitHub: #{gemspec.version}"
-      end
-    end
+    fetch_all_govuk_gems.group_by { |gem| gem["team"] }.each do |team, gems|
+      message = gems.filter_map do |gem|
+        repo_name = gem["app_name"]
+        rubygems_version = fetch_rubygems_version(repo_name)
+        if !rubygems_version.nil? &&
+          files_changed_since_tag(repo_name, "v" + rubygems_version).any? { |path| path_is_built_into_gem?(path) }
+          "  #{repo_name} has unreleased changes since v#{rubygems_version}"
+        end
+      end.join("\n")
```

Comment:
`#print_version_discrepancies` is responsible for a lot of things:
 - finding the gems which have unreleased changes
 - formatting the message
 - grouping by the team

We should probably move some of this logic to separate methods.

Some of the patterns in this class don’t follow the style guide – I’d suggest configuring https://github.com/alphagov/rubocop-govuk first.

Reply:
This is the big one, and I suggest I tackle this as a separate PR on its own, maybe even two separate ones, one splitting out the logics by responsibilities, and the other main one is the rubocop-govuk implementation?